api/grpcserver/v2alpha1/activation.go

```diff
@@ -0,0 +1,348 @@
+					return nil
+				}
+				return status.Error(codes.Internal, err.Error())
+			}
+		}
+	}
+}
+
+func toAtx(atx *types.VerifiedActivationTx) *spacemeshv2alpha1.ActivationV1 {
+	v1 := &spacemeshv2alpha1.ActivationV1{
+		Id:             atx.ID().Bytes(),
+		NodeId:         atx.SmesherID.Bytes(),
+		Signature:      atx.Signature.Bytes(),
+		PublishEpoch:   atx.PublishEpoch.Uint32(),
+		Sequence:       atx.Sequence,
+		PreviousAtx:    atx.PrevATXID[:],
+		PositioningAtx: atx.PositioningATX[:],
+		Coinbase:       atx.Coinbase.String(),
+		Units:          atx.NumUnits,
+		BaseHeight:     uint32(atx.BaseTickHeight()),
+		Ticks:          uint32(atx.TickCount()),
+	}
+	if atx.CommitmentATX != nil {
+		v1.CommittmentAtx = atx.CommitmentATX.Bytes()
+	}
+	if atx.VRFNonce != nil {
+		v1.VrfPostIndex = &spacemeshv2alpha1.VRFPostIndex{
+			Nonce: uint64(*atx.VRFNonce),
+		}
+	}
+	if atx.InitialPost != nil {
+		v1.InitialPost = &spacemeshv2alpha1.Post{
+			Nonce:   atx.InitialPost.Nonce,
+			Indices: atx.InitialPost.Indices,
+			Pow:     atx.InitialPost.Pow,
+		}
+	}
+	if nipost := atx.NIPost; nipost != nil {
+		if nipost.Post != nil {
+			v1.Post = &spacemeshv2alpha1.Post{
+				Nonce:   nipost.Post.Nonce,
+				Indices: nipost.Post.Indices,
+				Pow:     nipost.Post.Pow,
+			}
+		}
+		if nipost.PostMetadata != nil {
+			v1.PostMeta = &spacemeshv2alpha1.PostMeta{
+				Challenge:     nipost.PostMetadata.Challenge,
+				LabelsPerUnit: nipost.PostMetadata.LabelsPerUnit,
+			}
+		}
+		v1.Membership = &spacemeshv2alpha1.PoetMembershipProof{
+			ProofNodes: make([][]byte, len(nipost.Membership.Nodes)),
+			Leaf:       nipost.Membership.LeafIndex,
+		}
+		for i, node := range nipost.Membership.Nodes {
+			v1.Membership.ProofNodes[i] = node.Bytes()
+		}
+	}
+	return v1
+}
+
+func NewActivationService(db *sql.Database) *ActivationService {
+	return &ActivationService{db: db}
+}
+
+type ActivationService struct {
+	db *sql.Database
+}
+
+var _ grpcserver.ServiceAPI = (*ActivationService)(nil)
+
+func (s *ActivationService) RegisterService(server *grpc.Server) {
+	spacemeshv2alpha1.RegisterActivationServiceServer(server, s)
+}
+
+func (s *ActivationService) RegisterHandlerService(mux *runtime.ServeMux) error {
+	return spacemeshv2alpha1.RegisterActivationServiceHandlerServer(context.Background(), mux, s)
+}
+
+// String returns the service name.
+func (s *ActivationService) String() string {
+	return "ActivationService"
+}
+
+func (s *ActivationService) List(
+	ctx context.Context,
+	request *spacemeshv2alpha1.ActivationRequest,
+) (*spacemeshv2alpha1.ActivationList, error) {
+	ops, err := toOperations(request)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	// every full atx is ~1KB. 100 atxs is ~100KB.
+	if request.Limit > 100 {
+		return nil, status.Error(codes.InvalidArgument, "limit is capped at 100")
+	} else if request.Limit == 0 {
+		return nil, status.Error(codes.InvalidArgument, "limit must be set to <= 100")
+	}
```

Comment:
NIT: I find a switch more readable:
```suggestion
	switch {
	case request.Limit > 100:
		return nil, status.Error(codes.InvalidArgument, "limit is capped at 100")
	case request.Limit == 0:
		return nil, status.Error(codes.InvalidArgument, "limit must be set to <= 100")
	}
```